src/lib.rs

```diff
@@ -14,10 +14,10 @@ extern crate log;
 extern crate vm_memory;
 extern crate vmm_sys_util;
 
-mod device;
+mod devices;
 mod queue;
 
-pub use self::device::*;
+pub use self::devices::*;
```

Comment:
I think that queues and devices are conceptually different and can be used independently (at least the queues). For this reason, what do you say about exporting the modules directly instead of `pub use`?

My general thinking when using flattened vs non-flattened exports is that flattened exports are to be used when the modules are there just to provide a structural separation, and not a logical one.

**Structural separation**: If everything would be in one module, than the module would be to big, and thus the code becomes hard to follow, hence we are splitting it in multiple modules. This artificial separation is hidden from the end user by exporting everything flattened.
**Logical separation**: separate concepts are split between logical modules.